mobile/src/androidTest/java/ch/epfl/sdp/mobile/data/features/authentication/AlwaysSucceedingAuthenticationApi.kt

```diff
@@ -0,0 +1,48 @@
+package ch.epfl.sdp.mobile.data.features.authentication
+
+import ch.epfl.sdp.mobile.data.api.AuthenticationApi
+import ch.epfl.sdp.mobile.data.api.AuthenticationApi.AuthenticationResult.Success
+import ch.epfl.sdp.mobile.data.api.AuthenticationApi.User
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.map
+
+/**
+ * A fake implementation of [AuthenticationApi] which is backed by an internal state flow, and whose
+ * calls to [signInWithEmail] and [signUpWithEmail] will always succeed.
+ */
+class AlwaysSucceedingAuthenticationApi : AuthenticationApi {
```

Comment:
Why do we need an AlwaysSucceedingAuthenticationApi fake API ?

Reply:
We definitely could, but having a fake API implementation offers some flexibility when writing certains tests too. Is there something specific that would make mocks a more obvious choice ?

Reply:
Not really, I was just wondering about the reason that made you create a fake API over mocking 😀